### ISSUE TITLE
astal-tray icons not working properly

### ISSUE THREAD

**Issue by AlexL64** [Aylur/astal#165]
**Describe the bug**
Multiple tray items are either missing icons or displaying incorrect ones. These items have proper icons on AGSv1 or Waybar.

Below is a list of items missing icons:
- Mullvad VPN
- Lunar Client
- Vesktop
- GalaxyBudsClient

Additionally, the icon for Deezer is incorrect and does not match any existing icon.

**To Reproduce**
Launch one of the softwares listed above and run `astal-tray -d`.

**Expected behavior**
Every item has an icon, and it is the correct one.

**Additional context**
Here is the output from `astal-tray -d`
```
{
    "event": "item_added",
    "id": ":1.142/StatusNotifierItem",
    "item": {
        "item_id": ":1.142/StatusNotifierItem",
        "id": "chrome_status_icon_1",
        "bus_name": ":1.142",
        "object_path": "/StatusNotifierItem",
        "title": "",
        "status": "Active",
        "category": "ApplicationStatus",
        "tooltip": "Connected. Paris, France",
        "icon_theme_path": null,
        "icon_name": null,
        "menu_path": "/com/canonical/dbusmenu",
        "is_menu": false
    }
}
{
    "event": "item_added",
    "id": ":1.518/StatusNotifierItem",
    "item": {
        "item_id": ":1.518/StatusNotifierItem",
        "id": "chrome_status_icon_1",
        "bus_name": ":1.518",
        "object_path": "/StatusNotifierItem",
        "title": "",
        "status": "Active",
        "category": "ApplicationStatus",
        "tooltip": "Lunar Client",
        "icon_theme_path": null,
        "icon_name": null,
        "menu_path": "/com/canonical/dbusmenu",
        "is_menu": false
    }
}
{
    "event": "item_added",
    "id": "org.kde.StatusNotifierItem-105314-1/StatusNotifierItem",
    "item": {
        "item_id": "org.kde.StatusNotifierItem-105314-1/StatusNotifierItem",
        "id": "com.hunterwittenborn.Celeste",
        "bus_name": "org.kde.StatusNotifierItem-105314-1",
        "object_path": "/StatusNotifierItem",
        "title": "Celeste",
        "status": "Active",
        "category": "ApplicationStatus",
        "tooltip": "",
        "icon_theme_path": "",
        "icon_name": "com.hunterwittenborn.Celeste.CelesteTraySyncing-symbolic",
        "menu_path": "/MenuBar",
        "is_menu": false
    }
}
{
    "event": "item_added",
    "id": ":1.12/org/ayatana/NotificationItem/nm_applet",
    "item": {
        "item_id": ":1.12/org/ayatana/NotificationItem/nm_applet",
        "id": "nm-applet",
        "bus_name": ":1.12",
        "object_path": "/org/ayatana/NotificationItem/nm_applet",
        "title": "Network",
        "status": "Active",
        "category": "SystemServices",
        "tooltip": "",
        "icon_theme_path": "",
        "icon_name": "nm-signal-50-secure",
        "menu_path": "/org/ayatana/NotificationItem/nm_applet/Menu",
        "is_menu": false
    }
}
{
    "event": "item_added",
    "id": ":1.529/org/ayatana/NotificationItem/Deezer1",
    "item": {
        "item_id": ":1.529/org/ayatana/NotificationItem/Deezer1",
        "id": "Deezer1",
        "bus_name": ":1.529",
        "object_path": "/org/ayatana/NotificationItem/Deezer1",
        "title": "electron13",
        "status": "Active",
        "category": "ApplicationStatus",
        "tooltip": "",
        "icon_theme_path": "/tmp/.org.chromium.Chromium.zJRB99",
        "icon_name": "Deezer1_1",
        "menu_path": "/org/ayatana/NotificationItem/Deezer1/Menu",
        "is_menu": false
    }
}
{
    "event": "item_added",
    "id": ":1.133/StatusNotifierItem",
    "item": {
        "item_id": ":1.133/StatusNotifierItem",
        "id": "chrome_status_icon_1",
        "bus_name": ":1.133",
        "object_path": "/StatusNotifierItem",
        "title": "",
        "status": "Active",
        "category": "ApplicationStatus",
        "tooltip": "Vesktop",
        "icon_theme_path": null,
        "icon_name": null,
        "menu_path": "/com/canonical/dbusmenu",
        "is_menu": false
    }
}
{
    "event": "item_added",
    "id": ":1.500/org/ayatana/NotificationItem/steam",
    "item": {
        "item_id": ":1.500/org/ayatana/NotificationItem/steam",
        "id": "steam",
        "bus_name": ":1.500",
        "object_path": "/org/ayatana/NotificationItem/steam",
        "title": "Steam",
        "status": "Active",
        "category": "ApplicationStatus",
        "tooltip": "",
        "icon_theme_path": "/home/alex/.local/share/Steam/public",
        "icon_name": "steam_tray_mono",
        "menu_path": "/org/ayatana/NotificationItem/steam/Menu",
        "is_menu": false
    }
}
{
    "event": "item_added",
    "id": ":1.599/org/blueman/sni",
    "item": {
        "item_id": ":1.599/org/blueman/sni",
        "id": "blueman",
        "bus_name": ":1.599",
        "object_path": "/org/blueman/sni",
        "title": "blueman",
        "status": "Active",
        "category": "Hardware",
        "tooltip": "Bluetooth Disabled",
        "icon_theme_path": null,
        "icon_name": "blueman-disabled",
        "menu_path": "/org/blueman/sni/menu",
        "is_menu": false
    }
}
{
    "event": "item_added",
    "id": "org.kde.StatusNotifierItem-119841-0/StatusNotifierItem",
    "item": {
        "item_id": "org.kde.StatusNotifierItem-119841-0/StatusNotifierItem",
        "id": "Galaxy Buds",
        "bus_name": "org.kde.StatusNotifierItem-119841-0",
        "object_path": "/StatusNotifierItem",
        "title": "Galaxy Buds",
        "status": "Passive",
        "category": "ApplicationStatus",
        "tooltip": "",
        "icon_theme_path": "",
        "icon_name": "",
        "menu_path": "/net/avaloniaui/dbusmenu/c1fc9093ffb04749ac9ec364d9fa8a48",
        "is_menu": false
    }
}
```

**Comment by jordhan94** [comment]
the `TrayItem` comes with a `Gio.Icon` prop, here is an example:

```
bind(tray, "items").subscribe((items: Tray.TrayItem[]) => {
    ...
    items.map((item: Tray.TrayItem) => {
        ...
        return new Widget.EventBox({
            ...
            // here you can create a icon widget
            child: new Widget.Icon({ gIcon: item.gicon }),
    });
    ...
});
```

**Comment by koeqaife** [comment]
icon = Gtk.Image()
icon.set_pixel_size(18)
item.bind_property("gicon", icon, "gicon", SYNC)

I used Gtk.Image instead of Astal.Icon and it worked btw

**Comment by kotontrion** [comment]
please check if this fixes it #187.

It should fix the issue @koeqaife mentioned above.
Not sure about the missing tray icons, but as mullvad and vesktop work fine for me, it would indicate that these issues are indeed related.

**Comment by AlexL64** [comment]
Thanks for your help.

I realized I was doing it wrong. I was using the `icon` property instead of the `gicon` property, which caused certain items to have missing or incorrect icons.

I simply replaced `<icon icon={bind(item, "iconName")} />` with `<icon gIcon={bind(item, "gicon")} />`, and now it works.
